### PR TITLE
Damage System

### PR DESCRIPTION
**Pull request by Steelstone3**

### CODE REVIEW

File: src/systems/weapons/player_mine/player_mine_collision.rs [hunk 1]
```diff
@@ -35,6 +35,12 @@ pub fn player_mine_collision_with_starship(
                 mine.mine.weapon.damage.calculate_damage();
                 starship.take_damage(mine.mine.weapon.damage);
 
+                tracing::info!(
```
Comment (Steelstone3) [inline src/systems/weapons/player_mine/player_mine_collision.rs:38]: Not actually doing damage to the starship :man_shrugging:

File: src/components/weapons/lock_on_weapon.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+use super::ranged_weapon::RangedWeapon;
+use crate::components::starships::starship::Starship;
+use bevy::{ecs::component::Component, transform::components::Transform};
+
+// TODO idea for targetting system
+#[derive(Component, Debug, PartialEq)]
+pub struct LockOnWeapon {
```
Comment (Steelstone3) [inline src/components/weapons/lock_on_weapon.rs:7]: What do you think of this idea for a possible path to go down for weapon targetting. On starship there is a method to check if it "is_destroyed" which you can use to remove the targetting square. The translation in transform can be used to get the current position of the starship.

File: src/components/weapons/weapon.rs [hunk 3]
```diff
@@ -1,23 +1,20 @@
-use bevy::{
-    ecs::component::Component,
-    math::{Vec2, Vec3},
-};
+use bevy::{ecs::component::Component, math::Vec2};
+
+use super::damage::Damage;
 
 #[derive(Component, Debug, PartialEq)]
 pub struct Weapon {
```
Comment (Steelstone3) [inline src/components/weapons/weapon.rs:6]: This now acts as a "Base" weapon that can be extended

File: src/components/weapons/player_mine.rs [hunk 4]
```diff
@@ -22,9 +26,17 @@ mod player_mine_should {
                 mine: Default::default(),
                 firing_sound: Default::default(),
                 impact_sound: Default::default(),
-                velocity: -5.0,
-                size: Vec2 { x: 100.0, y: 100.0 },
-                lifetime: Timer::from_seconds(10.0, TimerMode::Once),
+                lifetime_weapon: LifetimeWeapon {
```
Comment (Steelstone3) [inline src/components/weapons/player_mine.rs:29]: Lifetime weapons despawn after a timer

**Review by MatthewGriffin — APPROVED** [review]

**Pull request merged**